pkg/proxy/endpointslicecache.go

```diff
@@ -214,16 +219,15 @@ func formatEndpointsList(endpoints []Endpoint) []string {
 }
 
 // endpointSliceCacheKeys returns cache keys used for a given EndpointSlice.
-func endpointSliceCacheKeys(endpointSlice *discovery.EndpointSlice) (types.NamespacedName, string) {
-	if len(endpointSlice.OwnerReferences) == 0 {
-		klog.Errorf("No owner reference set on endpoint slice: %s", endpointSlice.Name)
-		return types.NamespacedName{}, endpointSlice.Name
-	}
-	if len(endpointSlice.OwnerReferences) > 1 {
-		klog.Errorf("More than 1 owner reference set on endpoint slice: %s", endpointSlice.Name)
+func endpointSliceCacheKeys(endpointSlice *discovery.EndpointSlice) (types.NamespacedName, string, error) {
+	var err error
+	serviceName, ok := endpointSlice.Labels[discovery.LabelServiceName]
+	if !ok || serviceName == "" {
+		err = fmt.Errorf("No %s label set on endpoint slice: %s", discovery.LabelServiceName, endpointSlice.Name)
+	} else if endpointSlice.Namespace == "" || endpointSlice.Name == "" {
+		err = fmt.Errorf("Expected EndpointSlice name and namespace to be set: %v", endpointSlice)
```

Comment:
Should nil be returned as the first component of return value (types.NamespacedName) in this case ?

Reply:
See #82389